src/utils/custom-highlight-lang/swift.js

```diff
@@ -41,5 +41,40 @@ export default function swiftOverride(hljs) {
     };
   }
 
+  // Checks if a given language sub-mode matches the "ESCAPED_NEWLINE" from the
+  // built-in Swift parser from hljs
+  const isEscapedNewlineMode = (mode) => {
+    const { className, match } = mode;
+    if (className !== 'subst' || !match) {
+      return false;
+    }
+
+    const matchStr = match.toString();
+    return matchStr.startsWith('\\') && matchStr.endsWith('[\\t ]*(?:[\\r\\n]|\\r\\n)');
+  };
+
+  // replace the "ESCAPED_NEWLINE" sub-mode in the multiline string literal mode
+  // variants so that it doesn't include the actual newline characters in the
+  // span token that it generates, because this causes issues with our
+  // line-number + multi-line string literal logic when the span for the
+  // backslash token is split across multiple lines
+  const strIndex = language.contains.findIndex(({ className }) => className === 'string');
+  language.contains[strIndex] = {
+    ...language.contains[strIndex],
+    variants: language.contains[strIndex].variants.map(variant => ({
+      ...variant,
+      contains: variant.contains.map(mode => (isEscapedNewlineMode(mode) ? ({
+        className: 'subst',
+        begin: /\\#{0,3}/,
```

Comment:
clever 🤔 should we add a test?

Reply:
I would, but I'm not familiar enough with the intricacies of this extended syntax to write a realistic sample 😅